packages/ketcher-core/src/application/editor/actions/paste.ts

```diff
@@ -84,6 +89,13 @@ export function fromPaste(
 
   pstruct.frags.forEach((frag, frid) => {
     if (!frag) return;
+    if (frag.properties) {
+      action.addOp(
```

Comment:
These are needed b/c on paste tool mousemove event the fragments are re-inserted into the struct and all their properties are lost.